src/components/Shared/StringFileUploadField/index.jsx

```diff
@@ -74,95 +88,123 @@ const StringFileUploadField = ({
       handleFile(file);
     }
   };
 
+  const fieldsRef = useRef(null);
+  const [fieldsDivHeight, setFieldsDivHeight] = useState(0);
+
+  // Keep track of the computed height
+  useEffect(() => {
+    if (fieldsRef.current) {
+      setFieldsDivHeight(fieldsRef.current.offsetHeight);
+    }
+  }, []);
+
   return (
-    <Box>
-      <Grid2 container direction="row" spacing={3} alignItems="center">
-        <Grid2 size="grow">
-          <TextField
-            type="url"
-            id="standard-full-width"
-            label={labelKeyword}
-            placeholder={placeholderKeyword}
-            fullWidth
-            value={urlInput}
-            variant="outlined"
-            disabled={isParentLoading || fileInput instanceof Blob}
-            onChange={(e) => setUrlInput(e.target.value)}
-          />
-        </Grid2>
-        <Grid2>
-          <LoadingButton
-            type="submit"
-            variant="contained"
-            color="primary"
-            onClick={async (e) => {
-              e.preventDefault();
-              urlInput ? await handleSubmit(urlInput) : await handleSubmit(e);
-            }}
-            loading={isParentLoading}
-            disabled={(urlInput === "" && !fileInput) || isParentLoading}
-          >
-            {submitButtonKeyword}
-          </LoadingButton>
-        </Grid2>
-      </Grid2>
-      <Grid2 mt={2}>
-        <ButtonGroup
-          variant="outlined"
-          disabled={isParentLoading || urlInput !== ""}
-        >
-          <Button
-            startIcon={<FolderOpenIcon />}
-            sx={{ textTransform: "none" }}
-            style={
-              isOver ? { cursor: validDrop ? "copy" : "no-drop" } : undefined
-            }
-            onDragEnter={onDragEnter}
-            onDragLeave={onDragLeave}
-            onDragOver={(e) => e.preventDefault()}
-            onDrop={onDrop}
-          >
-            <label htmlFor="file">
-              {fileInput ? fileInput.name : localFileKeyword}
-            </label>
-            <input
-              id="file"
-              name="file"
-              type="file"
-              accept={fileInputTypesAccepted}
-              hidden={true}
-              ref={fileRef}
-              onChange={(e) => {
-                e.preventDefault();
-                handleFile(e.target.files[0]);
-                e.target.value = null;
-              }}
+    <Box
+      onDragOver={onDragOver}
+      onDragLeave={onDragLeave}
+      onDrop={onDrop}
+      sx={{
+        border: isDragging ? "4px dashed #00926c" : "0",
+        height: isDragging ? fieldsDivHeight : "auto",
+        backgroundColor: isDragging ? dropColor : "initial",
+      }}
+    >
+      {isDragging && (
+        <Stack justifyContent="center" alignItems="center" height="100%">
+          <Typography>{keyword("droppable_zone")}</Typography>
+        </Stack>
+      )}
+
+      <Box visibility={isDragging ? "hidden" : "visible"} ref={fieldsRef}>
```

Comment:
You can probably get away without the `useRef` and the effect if you do something like

```javascript
    <Box
      onDragOver={onDragOver}
      onDragLeave={onDragLeave}
      onDrop={onDrop}
      sx={{
        position: "relative",
      }}
    >
      <Box>
        {/* form content goes here */}
      </Box>

      {isDragging && (
        <Stack
          sx={{
            border: "4px dashed #00926c",
            backgroundColor: dropColor,
            justifyContent: "center",
            alignItems: "center",
            position: "absolute",
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
          }}
        >
          <Typography>{keyword("droppable_zone")}</Typography>
        </Stack>
      )}
    </Box>
```

This uses absolute positioning to put the drop zone box on top of the form fields area, and it doesn't need the effect to track the height as it gets that automatically with t/l/b/r all set to zero.  To get this to work I've had to move the Stack after the Box, if you leave them the other way round you might need a `zIndex` on the Stack to make it sit on top.